modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/datatypes/NumericCaseTypeCoercionTest.java

```diff
@@ -94,40 +95,38 @@ public void numericWithLiteralsCoercion(
             Matcher<RexNode> firstOperandMatcher,
             Matcher<RexNode> secondOperandMatcher
     ) throws Exception {
+        // https://issues.apache.org/jira/browse/IGNITE-23088
+        // break serialization check otherwise, in: 9223372034707302160, out: 9.223372034707302E18
+        NativeTypeSpec firstType = typePair.second().spec();
+        // if approximate numeric
+        if (firstType == NativeTypeSpec.FLOAT || firstType == NativeTypeSpec.DOUBLE) {
+            NativeTypeSpec secondType = typePair.first().spec();
+            // if exact numeric
+            if (secondType == NativeTypeSpec.INT8 || secondType == NativeTypeSpec.INT16 || secondType == NativeTypeSpec.INT32
+                    || secondType == NativeTypeSpec.INT64) {
+                return;
+            }
+        }
+
         List<Object> params = List.of(
-                generateValueByType(typePair.first()), generateValueByType(typePair.second())
+                generateLiteralWithNoRepeatitio(typePair.first()), generateLiteralWithNoRepeatitio(typePair.second())
         );
 
         assertPlan(format("SELECT CASE WHEN RAND_UUID() != RAND_UUID() THEN {} ELSE {} END FROM t", params.get(0), params.get(1)),
                 SCHEMA, operandCaseMatcher(firstOperandMatcher, secondOperandMatcher)::matches, List.of());
     }
 
-    private static Object generateValueByType(NativeType type) {
-        increment++;
-
-        if (type.equals(NativeTypes.INT8) || type.equals(NativeTypes.INT16) || type.equals(NativeTypes.INT32)) {
-            return SqlTestUtils.generateValueByType(increment % 2, type.spec().asColumnType());
-        } else if (type.equals(NativeTypes.INT64)) {
-            // break serialization check otherwise, in: 9223372034707302160, out: 9.223372034707302E18
-            return (long) SqlTestUtils.generateValueByType(Integer.MAX_VALUE - ((increment % 2) * 10000),
-                    type.spec().asColumnType()) / 1000;
-        } else if (type.equals(NativeTypes.FLOAT)) {
-            float res = (float) SqlTestUtils.generateValueByType(type.spec().asColumnType());
-            if (increment % 2 == 1) {
-                // decrease exponent part
-                res /= 10;
-            }
-            return res;
-        } else if (type.equals(NativeTypes.DOUBLE)) {
-            double res = (double) SqlTestUtils.generateValueByType(type.spec().asColumnType());
-            if (increment % 2 == 1) {
-                // decrease exponent part
-                res /= 10;
-            }
-            return res;
-        } else {
-            return SqlTestUtils.generateValueByType(increment % 2, type.spec().asColumnType());
-        }
+    private static String prevResult = "";
+
+    // It is necessary to have a guarantee that two values in a row will be different.
+    private static String generateLiteralWithNoRepeatitio(NativeType type) {
```

Comment:
Typo in naming, plus may be it more useful to make a function with varlen types params which returns List of nonrepeating results ? wdyt ?

Reply:
done.